### PR TITLE
Nev man1 food

### PR DESCRIPTION
**Pull request by NevMan1**
This is a new  feature that I implemented. It also utilizes the Gemini AI model. It looks up what meals are in the dining halls and recommends the food that would be good for you.
Acceptance Criteria

- You  are able to generate meal plan
- The prompt shows up
- You can close out of the tab
-

### CODE REVIEW

File: PhoneAuth/.swift
Comment (brycewangg) [inline PhoneAuth/.swift:1]: Is this file needed?

File: PhoneAuth/Controllers/MealPlanViewController.swift [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+                                self?.showPopup(message: message)
+                            } else {
+                                self?.showPopup(message: "Error: \(message)")
+                            }
+                        
+                        
+                    }
+                    
+                case .failure(let error):
+                    self.resultLabel.text = "Error: \(error.localizedDescription)"
```
Comment (brycewangg) [inline PhoneAuth/Controllers/MealPlanViewController.swift]: Is this error thrown when dining commons are closed? It should display that as the error.

File: PhoneAuth/.swift
Comment (NevMan1) [reply PhoneAuth/.swift:1]: I don't know honestly

File: PhoneAuth/Controllers/MealPlanViewController.swift (hunk 1 above)
Comment (NevMan1) [reply PhoneAuth/Controllers/MealPlanViewController.swift]: It doesn't say it is closed, all it says is that the data couldn't be read because it wasn't in the correct format

File: PhoneAuth/.swift
Comment (Tommygithubaccount123) [reply PhoneAuth/.swift:1]: I'd say remove it from the branch

File: PhoneAuth/Controllers/MealPlanViewController.swift (hunk 1 above)
Comment (brycewangg) [reply PhoneAuth/Controllers/MealPlanViewController.swift]: instead of displaying "Error: \(error.localizedDescription)" you can just display something like "Dining halls are closed" cause i think thats the only time the error occurs

File: PhoneAuth/UCSB.plist
Comment (Tommygithubaccount123) [inline PhoneAuth/UCSB.plist:1]: remove this file, get a new api key if it matters

File: PhoneAuth/Controllers/MealPlanViewController.swift (hunk 1 above)
Comment (Tommygithubaccount123) [reply PhoneAuth/Controllers/MealPlanViewController.swift]: This error pops up when the dining common is closed I think. Can the popup show a msg for a user and not a developer? Ex: "Could not create meal since dining commons are closed"
